ingenialink/pdo.py

```diff
@@ -306,107 +321,80 @@ def reset_tpdo_mapping(self) -> None:
             self.write(map_register, 0)
         self._tpdo_maps.clear()
 
-    def add_rpdo_map(self, rpdo_map: RPDOMap) -> None:
-        """Add a new RPDO map into the drive.
-
-        It takes the first available (not mapped yet) RPDO assignment slot of the drive.
-
-        Args:
-            rpdo_map: RPDO map to be added.
+    def map_rpdos(self) -> None:
+        """Map the RPDO registers into the servo drive.
+        It takes the first available RPDO assignment slot of the drive.
 
         Raises:
             ILError: If there are no available PDOs.
         """
-        if len(self._rpdo_maps) + 1 > self.AVAILABLE_PDOS:
-            raise ILError("Could not add a new RPDO map, there are no available PDOs.")
-        self._rpdo_maps.append(rpdo_map)
-        map_index = len(self._rpdo_maps) - 1
-
+        if len(self._rpdo_maps) > self.AVAILABLE_PDOS:
```

Comment:
I think you should create a variable to store the number of RPDO maps.

Reply:
Why have more complexity and maintenance?
If it's a worry about performance read this: https://stackoverflow.com/questions/699177/python-do-python-lists-keep-a-count-for-len-or-does-it-count-for-each-call